src/domain/dataset_bert.py

```diff
@@ -112,7 +118,8 @@ def get_values(self, set_, part_, dataset, tokenizer):
         Returns:
             list: return a list of tokenized documents
         """
-        return [self.encode_document(document, tokenizer) for document in dataset[set_][part_]]
+        return Parallel(n_jobs=4)(delayed(encode_document)(document, tokenizer) for document in dataset[set_][part_])
```

Comment:
```suggestion
        return Parallel(n_jobs=-1)(delayed(encode_document)(document, tokenizer) for document in dataset[set_][part_])
```

Reply:
What happens when encode_document encounters a None document (as in the None check it contains) ? 

Wouldn't this method then return a list possibly containing None ? i.e. [encoded_doc1, ..., None, ...] ? Doesn't this mess up the rest of the training process ? 

My personal suggestion:

`return Parallel(n_jobs=-1)(delayed(encode_document)(document, tokenizer) for document in dataset[set_][part_] if document)`

The `if document` filters out empty documents before parallelization to go around this possible bug.